libraries/shared/logs/extractor.go

```diff
@@ -56,26 +72,42 @@ func (extractor *LogExtractor) AddTransformerConfig(config transformer.EventTran
 		return checkedHeadersErr
 	}
 
-	if extractor.StartingBlock == nil {
-		extractor.StartingBlock = &config.StartingBlockNumber
-	} else if earlierStartingBlockNumber(config.StartingBlockNumber, *extractor.StartingBlock) {
+	if resetStartingBlockNumber(config.StartingBlockNumber, extractor.StartingBlock) {
 		extractor.StartingBlock = &config.StartingBlockNumber
 	}
 
+	if resetEndingBlockNumber(config.EndingBlockNumber, extractor.EndingBlock) {
+		extractor.EndingBlock = &config.EndingBlockNumber
+	}
+
 	addresses := transformer.HexStringsToAddresses(config.ContractAddresses)
 	extractor.Addresses = append(extractor.Addresses, addresses...)
 	extractor.Topics = append(extractor.Topics, common.HexToHash(config.Topic))
 	return nil
 }
 
+func resetStartingBlockNumber(currentTransformerBlock int64, extractorBlock *int64) bool {
```

Comment:
similarly very minor but I wonder if we should name these in a way that indicates they're checking a condition rather than actually resetting anything?